tools/setup_helpers/generate_code.py

```diff
@@ -1,9 +1,16 @@
 import argparse
 import os
+import pathlib
+import shutil
 import sys
+import tempfile
 import yaml
 from typing import Any, List, Optional, cast
 
+# Note that even though this is deprecated, a certain Meta internal
+# build is not compatible with importlib.resources file listing.
+import pkg_resources
```

Comment:
This import is very expensive https://github.com/pytorch/pytorch/pull/71902 can we do without it whenever possible?

Reply:
I think that argument makes sense for the library we produce, but I don't think it makes much sense for our code generation. Does anything import this other than the build?

Also, this import isn't that slow on my crummy personal laptop:
```
$ hyperfine --warmup=2 $'python -c \'import pkg_resources\'' $'python -c \'from pkg_resources import packaging\''
Benchmark 1: python -c 'import pkg_resources'
  Time (mean ± σ):     167.8 ms ±  25.1 ms    [User: 139.0 ms, System: 14.0 ms]
  Range (min … max):   127.7 ms … 217.0 ms    18 runs
 
Benchmark 2: python -c 'from pkg_resources import packaging'
  Time (mean ± σ):     148.5 ms ±  24.4 ms    [User: 127.9 ms, System: 13.8 ms]
  Range (min … max):   117.6 ms … 186.8 ms    18 runs
 
Summary
  'python -c 'from pkg_resources import packaging'' ran
    1.13 ± 0.25 times faster than 'python -c 'import pkg_resources''
```

Reply:
> Also, this import isn't that slow on my crummy personal laptop

IIRC it depends a lot of the software stack and configuration. It is not universally slow.

I agree it is not a hard blocker but our codegen takes ~10/20s to run. So if that takes 1s to import, it is a significant slowdown.
That's why I was asking if there is a way to have a simple guard to not use this when possible?

Reply:
So, pkg_resources is used unconditionally by main. Moving the import lower would help if anything were to import this as a module. `git grep import.*generate_code` shows no results as of today.

Reply:
To be frank, I don't like the idea myself. Can we workaround it somehow (working on a similar change for gen.pyi). Can't we just use filegroups and add those files as package sources?

Reply:
Will that work the same way in Bazel and Buck? If you look at the internal diff corresponding to this, I think it was done somewhat like you suggest but that ended up needing different code paths between Buck and Bazel.

What are your objections specifically to this?